File: Formularios/infoagregaitem.cs
using System;
using System.Globalization;
using System.Windows.Forms;

namespace Centrex
{
    public partial class infoagregaitem
    {

        private bool produccion = false;
        private bool ordenCompra = false;
        private bool comprobanteCompra = false;
        private int id_comprobanteCompra = -1;
        private bool actualiza = true;
        private int idUsuario;
        private Guid idUnico;
        public int cant;

        public infoagregaitem()
        {
            InitializeComponent();
        }

        public infoagregaitem(int _idUsuario, Guid _idUnico)
        {
            InitializeComponent();
            idUsuario = _idUsuario;
            idUnico = _idUnico;
        }

        public infoagregaitem(bool _produccion, bool _ordenCompra, bool _actualiza, int _idUsuario, Guid _idUnico)
        {
            InitializeComponent();
            produccion = _produccion;
            ordenCompra = _ordenCompra;
            actualiza = _actualiza;
            idUsuario = _idUsuario;
            idUnico = _idUnico;
        }

        public infoagregaitem(bool _comprobanteCompra, int _id_comprobanteCompra)
        {
            InitializeComponent();
            comprobanteCompra = _comprobanteCompra;
            id_comprobanteCompra = _id_comprobanteCompra;
        }

        private void infoagregaitem_FormClosed(object sender, FormClosedEventArgs e)
        {
        }

        private void infoagregaitem_Load(object sender, EventArgs e)
        {
            lbl_item.Text = edita_item.Item;
            lbl_desc.Text = edita_item.Descript;
            lbl_stock.Text = edita_item.Cantidad.Value.ToString();

            if (produccion)
                txt_precio.Enabled = false;

            if (agregaitem == true)
            {
                txt_cantidad.Text = 1.ToString();
                txt_precio.Text = edita_item.PrecioLista.ToString();
            }
            else if (produccion)
            {
                txt_cantidad.Text = producciones.askCantidadCargadaProduccion(edita_item.IdItem, -1, edita_item.IdItemTemporal).ToString();
            }
            else if (ordenCompra)
            {
                txt_cantidad.Text = ordenesCompras.askCantidadCargadaOC(edita_item.IdItem, id_tmpOCItem: edita_item.IdItemTemporal).ToString();
                txt_precio.Text = ordenesCompras.askPrecioCargadoOC(edita_item.IdItem, id_tmpOCItem: edita_item.IdItemTemporal).ToString();
            }
            else if (comprobanteCompra)
            {
            }
            else
            {
                txt_cantidad.Text = Pedidos.askCantidadCargada(edita_item.IdItem, -1, "tmppedidos_items", idUsuario, idUnico).ToString();
                txt_precio.Text = Pedidos.askPreciocargado(edita_item.IdItem, -1, "tmppedidos_items", idUsuario, idUnico).ToString();
            }
        }

        private void cmd_ok_Click(object sender, EventArgs e)
        {
            if (string.IsNullOrEmpty(txt_cantidad.Text))
                txt_cantidad.Text = 0.ToString();

            int cantidadInt = int.Parse(txt_cantidad.Text);
            if (cantidadInt == -1 || cantidadInt == 0)
            {
                closeandupdate(this);
                return;
            }

            if (double.Parse(txt_cantidad.Text) == 0d)
            {
                MessageBox.Show("La cantidad ingresada no puede ser 0, ingrese nuevamente", "Centrex", MessageBoxButtons.OK, MessageBoxIcon.Exclamation);
                return;
            }

            if (!actualiza)
            {
                cant = int.Parse(txt_cantidad.Text);
                Dispose();
                return;
            }

            if (!produccion && !ordenCompra && !comprobanteCompra)
            {
                double cantidadDisponible = edita_item.Cantidad.Value - double.Parse(txt_cantidad.Text);
                if (cantidadDisponible < 0)
                {
                    MessageBox.Show("No hay " + txt_cantidad.Text + " " + edita_item.Item.ToString() + " hay solo " + edita_item.Cantidad.ToString() + ".", "Centrex", MessageBoxButtons.OK, MessageBoxIcon.Exclamation);
                }
            }

            if (produccion)
            {
                // Es un pedido de produccion
                if (!agregaitem)
                {
                    addItemProducciontmp(ConvertToItem(edita_item), int.Parse(txt_cantidad.Text), edita_item.IdItemTemporal);
                }
                else
                {
                    producciones.addItemProducciontmp(ConvertToItem(edita_item), int.Parse(txt_cantidad.Text));
                }
            }
            else if (ordenCompra)
            {
                // Es una orden de compra
                if (!agregaitem)
                {
                    ordenesCompras.addItemOCtmp(ConvertToItem(edita_item), int.Parse(txt_cantidad.Text), decimal.Parse(txt_precio.Text, CultureInfo.CurrentCulture), edita_item.IdItemTemporal);
                }
                else
                {
                    ordenesCompras.addItemOCtmp(ConvertToItem(edita_item), int.Parse(txt_cantidad.Text), decimal.Parse(txt_precio.Text));
                }
            }
            else if (comprobanteCompra)
            {
                // Es un comprobante de compra
                comprobantes_compras.add_item_comprobanteCompra(id_comprobanteCompra, edita_item.IdItem, int.Parse(txt_cantidad.Text), decimal.Parse(txt_precio.Text));
            }
            // Pedido normal
            else if (!agregaitem)
            {
                Pedidos.AddItemPedidoTmp(edita_item, int.Parse(txt_cantidad.Text), decimal.Parse(txt_precio.Text), idUsuario, idUnico, edita_item.IdItemTemporal);
            }
            else
            {
                Pedidos.AddItemPedidoTmp(edita_item, int.Parse(txt_cantidad.Text), decimal.Parse(txt_precio.Text), idUsuario, idUnico);
            }

            Dispose();
        }

        private void cmd_exit_Click(object sender, EventArgs e)
        {
            Dispose();
        }

        private void txt_cantidad_KeyPress(object sender, KeyPressEventArgs e)
        {
            if (!generales.esNumero(e))
            {
                e.KeyChar = '\0';
            }
        }
    }
}

File: __Viejo/Formularios/infoagregaitem.vb
Public Class infoagregaitem

    Dim seleccionado As String
    Dim produccion As Boolean = False
    Dim ordenCompra As Boolean = False
    Dim comprobanteCompra As Boolean = False
    Dim id_comprobanteCompra As Integer = -1
    Dim actualiza As Boolean = True
    Dim idUsuario As Integer
    Dim idUnico As String
    Public cant As Integer

    Public Sub New()

        ' Esta llamada es exigida por el diseñador.
        InitializeComponent()

        ' Agregue cualquier inicialización después de la llamada a InitializeComponent().

    End Sub

    Sub New(_idUsuario As Integer, _idUnico As String)

        ' Esta llamada es exigida por el diseñador.
        InitializeComponent()

        ' Agregue cualquier inicialización después de la llamada a InitializeComponent().
        idUsuario = _idUsuario
        idUnico = _idUnico
    End Sub

    Public Sub New(ByVal _produccion As Boolean, ByVal _ordenCompra As Boolean, ByVal _actualiza As Boolean, ByVal _idUsuario As Integer, ByVal _idUnico As String)

        ' Esta llamada es exigida por el diseñador.
        InitializeComponent()

        ' Agregue cualquier inicialización después de la llamada a InitializeComponent().
        produccion = _produccion
        ordenCompra = _ordenCompra
        actualiza = _actualiza
        idUsuario = _idUsuario
        idUnico = _idUnico
    End Sub

    Public Sub New(ByVal _comprobanteCompra As Boolean, ByVal _id_comprobanteCompra As Integer)

        ' Esta llamada es exigida por el diseñador.
        InitializeComponent()

        ' Agregue cualquier inicialización después de la llamada a InitializeComponent().
        comprobanteCompra = _comprobanteCompra
        id_comprobanteCompra = _id_comprobanteCompra
        'actualiza = _actualiza
    End Sub

    Private Sub infoagregaitem_FormClosed(sender As Object, e As FormClosedEventArgs) Handles Me.FormClosed
        'closeandupdate(Me)
    End Sub

    Private Sub infoagregaitem_Load(sender As Object, e As EventArgs) Handles MyBase.Load

        lbl_item.Text = edita_item.item
        lbl_desc.Text = edita_item.descript
        lbl_stock.Text = edita_item.cantidad

        If produccion Then txt_precio.Enabled = False

        If agregaitem = True Then
            txt_cantidad.Text = 1
            txt_precio.Text = edita_item.precio_lista
        Else
            If produccion Then
                txt_cantidad.Text = askCantidadCargadaProduccion(edita_item.id_item,, edita_item.id_item_temporal)
            ElseIf ordenCompra Then
                txt_cantidad.Text = askCantidadCargadaOC(edita_item.id_item,, edita_item.id_item_temporal)
                txt_precio.Text = askPrecioCargadoOC(edita_item.id_item,, edita_item.id_item_temporal)
            ElseIf comprobanteCompra Then

            Else
                txt_cantidad.Text = askCantidadCargada(edita_item.id_item, -1, "tmppedidos_items", idUsuario, idUnico)
                txt_precio.Text = askPreciocargado(edita_item.id_item, -1, "tmppedidos_items", idUsuario, idUnico)
            End If
        End If
    End Sub

    Private Sub cmd_ok_Click(sender As Object, e As EventArgs) Handles cmd_ok.Click
        If txt_cantidad.Text = "" Then txt_cantidad.Text = 0
        If CInt(txt_cantidad.Text) = -1 Or CInt(txt_cantidad.Text) = 0 Then
            closeandupdate(Me)
            Exit Sub
        End If

        If txt_cantidad.Text = 0 Then
            MsgBox("La cantidad ingresada no puede ser 0, ingrese nuevamente", vbExclamation, "Centrex")
            Exit Sub
        End If

        If Not actualiza Then
            cant = txt_cantidad.Text
            Me.Dispose()
            Exit Sub
        End If

        If Not produccion And Not ordenCompra And Not comprobanteCompra Then
            If edita_item.cantidad - txt_cantidad.Text < 0 Then
                MsgBox("No hay " + txt_cantidad.Text + " " + edita_item.item.ToString + " hay solo " + edita_item.cantidad.ToString + ".", vbExclamation)
                'MsgBox("No hay " + txt_cantidad.Text + " " + edita_item.item.ToString + " hay solo " + edita_item.cantidad.ToString + ", ingrese una nueva cantidad o cancele", vbExclamation)
                'Exit Sub
            End If
        End If

        If produccion Then
            'Es un pedido de produccion
            If Not agregaitem Then
                addItemProducciontmp(edita_item, txt_cantidad.Text, edita_item.id_item_temporal)
            Else
                addItemProducciontmp(edita_item, txt_cantidad.Text)
            End If
        ElseIf ordenCompra Then
            'Es una orden de compra
            If Not agregaitem Then
                addItemOCtmp(edita_item, txt_cantidad.Text, txt_precio.Text, edita_item.id_item_temporal)
            Else
                addItemOCtmp(edita_item, txt_cantidad.Text, txt_precio.Text)
            End If
        ElseIf comprobanteCompra Then
            'Es un comprobante de compra
            add_item_comprobanteCompra(id_comprobanteCompra, edita_item.id_item, txt_cantidad.Text, txt_precio.Text)
        Else
            'Pedido normal
            If Not agregaitem Then
                addItemPedidotmp(edita_item, txt_cantidad.Text, txt_precio.Text, idUsuario, idUnico, edita_item.id_item_temporal)
            Else
                addItemPedidotmp(edita_item, txt_cantidad.Text, txt_precio.Text, idUsuario, idUnico)
            End If
        End If

        Me.Dispose()
    End Sub

    Private Sub cmd_exit_Click(sender As Object, e As EventArgs) Handles cmd_exit.Click
        Me.Dispose()
    End Sub

    Private Sub txt_cantidad_KeyPress(sender As Object, e As KeyPressEventArgs) Handles txt_cantidad.KeyPress
        If esNumero(e) = 0 Then
            e.KeyChar = ""
        End If
    End Sub
End Class